html/semantics/forms/the-fieldset-element/accessibility/fieldset-div-display-contents-manual.html

```diff
@@ -0,0 +1,12 @@
+<!doctype html>
+<title>fieldset accessibility test: fieldset div display: contents</title>
+<style>
+ div { display: contents; }
+</style>
+<fieldset>
+ <div>
+  <legend>Foo</legend>
```

Comment:
According to HTML spec [1], legend has to be a first child of fieldset. So you're right the browsers shouldn't pick up accessible name from a legend, which is not a direct child. I'd be nice to have a comment in the test or extend <title> to reflect this.

[1] https://html.spec.whatwg.org/multipage/form-elements.html#the-legend-element

Reply:
That is an authoring conformance requirement. The UA conformance requirement is

https://w3c.github.io/html-aam/#fieldset-element-accessible-name-computation

step 2